### ISSUE TITLE
mac-virtualcam emits debug logs into output of command line program using CMIO

### ISSUE THREAD

**Issue by stuart-elgato** [obsproject/obs-studio#8858]
### Operating System Info

macOS 13

### OBS Studio Version

29.1.0

### OBS Studio Log URL

OBS need only be installed, does not have to run

### Expected Behavior

I have a command line tool which browses the system for cameras, using the CMIO interfaces. The first call it makes asks the kCMIOObjectSystemObject for its devices, using this call:

CMIOObjectPropertyAddress address {
        .mSelector = kCMIOHardwarePropertyDevices,
        .mScope = kCMIOObjectPropertyScopeGlobal,
        .mElement = kCMIOObjectPropertyElementMain
    };

static const CMIOObjectID systemObjectID = kCMIOObjectSystemObject;
if (CMIOObjectHasProperty(systemObjectID, &address)) {

at this point, my tool should produce no output on std::cout

### Current Behavior

these strings are emitted by my tool (somehow, they got into my tool's std::cout)

Changed waitThread to realtime priority!
 mac-virtualcam(DAL): PlugInMain version=1.3.0
 mac-virtualcam(DAL): HardwarePlugIn_QueryInterface 
 mac-virtualcam(DAL): HardwarePlugIn_Release sRefCount now = 0
 mac-virtualcam(DAL): HardwarePlugIn_InitializeWithObjectID self=0x100f51638
 mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(36) kCMIOObjectPropertyListenerAdded self=0x100f51638 data(int)=1684629094
 mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
 mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(36) kCMIOObjectPropertyListenerAdded self=0x100f51638 data(int)=1869180523
 mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
 mac-virtualcam(DAL): HardwarePlugIn_ObjectSetPropertyData OBSDALDevice(36) kCMIOObjectPropertyListenerAdded self=0x100f51638 data(int)=1885762592
 mac-virtualcam(DAL): Device unhandled setPropertyDataWithAddress for kCMIOObjectPropertyListenerAdded
 mac-virtualcam(DAL): HardwarePlugIn_Teardown self=0x100f51638

### Steps to Reproduce

download the accompanying code:
[obs-tmi.zip](https://github.com/obsproject/obs-studio/files/11411058/obs-tmi.zip)

1. at the command line, run the tool supplied in the attached archive at executable/obs-tmi 
2. observe the output in Terminal

I also supplied an Xcode project which builds the tool, so you can see the single call to CMIOObjectHasProperty which causes the output.

i'm pretty sure it is OBS that is doing this. The strings include "OBSDALDevice", and this line:

kCMIOObjectPropertyListenerAdded self=0x100f51638 data(int)=1885762592

looks like it is emitted by line 267 of obs-studio/plugins/mac-virtualcam/src/dal-plugin/OBSDALPlugInInterface.mm

Note that I'm not objecting to the debug output, I would just expect it to go into a log file, not into the stdout of a tool which is accessing information about CMIO objects in general. The command line tool I am working on should be used in a CI/CD workflow, so we need to be able to parse its output. We don't expect strings in the output that we didn't put there.

**Comment by gxalpha** [comment]
These logs aren't printed to command line specifically, but sent to the Apple System Log facility using `NSLog` (in our code using the macros `DLog` and `DLogFunc`). This makes them appear in `Console.app`, but also in the command line of the app it's sent from. As the DAL plugin is loaded by the host application, that would be your TLI app. I'd consider this to work as expected.
That said, these messages should only appear if OBS (or the DAL plugin specifically) is build in Debug mode (which I assume is used here). Try running OBS in Release, configurations of the DAL plugin should not log anything (make sure that the DAL plugin gets replaced properly, if in doubt first delete `/Library/CoreMediaIO/Plug-Ins/DAL/obs-mac-virtualcam.plugin`).

**Comment by stuart-elgato** [comment]
I had recently installed OBS from the web site, but it seems I did have a debug version of the virtual cam at some point. I manually deleted the plugin, rebooted, installed OBS release from the web site, ran it and activated the virtual camera. Most of those logs don't appear any more, aside from "Changed waitThread to realtime priority!", which may come from some other plugin - I can't find that string or anything similar in OBS sources. Sorry for the noise, I'll close this issue.
